py/desispec/fiberflat.py

```diff
@@ -463,11 +441,11 @@ def compute_fiberflat(frame, nsig_clipping=10., accuracy=5.e-4, minval=0.1, maxv
     fiberflat = FiberFlat(wave, fiberflat, fiberflat_ivar, mask, mean_spectrum,
                      chi2pdf=chi2pdf,header=frame.meta,fibermap=frame.fibermap)
         
-    for broken_fiber in broken_fibers :
-        log.info("mask broken fiber {} in flat".format(broken_fiber))
-        fiberflat.fiberflat[fiber]=1.
-        fiberflat.ivar[fiber]=0.
-        fiberflat.mask[fiber]=specmask.BADFIBERFLAT
+    #for broken_fiber in broken_fibers :
+    #    log.info("mask broken fiber {} in flat".format(broken_fiber))
+    #    fiberflat.fiberflat[fiber]=1.
+    #    fiberflat.ivar[fiber]=0.
+    #    fiberflat.mask[fiber]=specmask.BADFIBERFLAT
```

Comment:
What if we modify the frame itself (like in other parts of desispec) rather just returning an altered ivar? Then the masks will be defined correctly down the line, wouldn't they? They wouldn't have the BADFIBERFLAT tag, but I don't believe that is applicable for these anyway now that we are correctly identifying blacklisted fibers as BADFIBER.

Reply:
I may be misunderstanding something, but in this case we are passing in a frame and computing a fiberflat to return, so the question is whether we are setting the fiberflat, ivar, and mask of the *fiberflat* object rather than the input frame.

In exposure 20200306/00053570 it looks like it failed to produce the fiberflat output file so I couldn't emperically check how the bits/ivar are set in the fiberflat for the bad fibers in the input.  Please check.

Reply:
@julienguy I may need your help in answering this question.  fiberflat is a derived quantity based on frames that are used to generate it. So if we modify the input frames to reflect proper ivar and mask, the derived fiberflats should have that, correct or incorrect?

Reply:
@julienguy   does your "good to me" include the code as it relates to Stephen's concern about fiberflats here?

Reply:
The routine `compute_fiberflat` does not copy the input frame mask to the output fiberflat. The fiberflat includes some smoothing and interpolation over masked flux. So the output fiberflat mask does not necessarily inherit all of the input frame specmask bits. but the fact it inherits the fibermask in the fibermap seems justified. The current code seems fine to me. Note the separate routine `mask_bad_fiberflat` that sets the bit `specmask.BADFIBERFLAT`. (and actually, as noted about by Stephen it should be `fiberflat.mask[fiber] |= specmask.BADFIBERFLAT` instead of `fiberflat.mask[fiber] = specmask.BADFIBERFLAT` in that routine.